sql/wsrep_applier.cc

```diff
@@ -176,6 +179,18 @@ int wsrep_apply_events(THD*        thd,
       }
     }
 #endif /* GTID_SUPPORT */
+    case QUERY_EVENT:
+      WSREP_DEBUG("query event: %s len: %d", ((Query_log_event*)ev)->query, ((Query_log_event*)ev)->q_len);
+      // Ugly hack coming next...
+      // We are going to explicitly start a transaction using
```

Comment:
Hmm, question here is why thd would be in active multi statement transaction at XA start ?